### ISSUE TITLE
Error compiling vterm-ctrl.c with clang on Mac, the issues don't look Mac-specific

### ISSUE THREAD

**Issue by jlokier** [akermu/emacs-libvterm#433]
Hi,

Here's an excerpt of the `*Install vterm*` Emacs buffer today, with two compilation errors and a notable warning. The errors look like a simple missing header file (for `strdup`). The warning looks like a genuine bug, a comparison that always returns false:

```
bin/vterm-ctrl.c:117:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(csi);
         ^
bin/vterm-ctrl.c:142:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(dcs);
         ^
bin/vterm-ctrl.c:129:10: warning: result of comparison of constant 156 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
    if(c == 0x9c) // ST
       ~ ^  ~~~~
```

These are installed software versions (probably irrelevant except clang):

```
Apple clang version 12.0.0 (clang-1200.0.26.2)
Target: x86_64-apple-darwin19.6.0
MacOS 10.15.7
libtool: stable 2.4.6 (bottled)  [with Homebrew]
cmake: stable 3.18.4 (bottled), HEAD  [with Homebrew] 
```

Here is the full copy of `*Install vterm*`:

```
-- The C compiler identification is AppleClang 12.0.0.12000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- No build type selected, defaulting to RelWithDebInfo
-- System libvterm not found: libvterm will be downloaded and compiled as part of the build process
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/jamie/.emacs.d/elpa/vterm-20201004.2057/build
Scanning dependencies of target libvterm
[  8%] Creating directories for 'libvterm'
[ 16%] Performing download step (git clone) for 'libvterm'
Cloning into 'libvterm'...
Note: switching to '65dbda3ed214f036ee799d18b2e693a833a0e591'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 65dbda3 Bump version to 0.1.3
[ 25%] Performing update step for 'libvterm'
[ 33%] No patch step for 'libvterm'
[ 41%] No configure step for 'libvterm'
[ 50%] Performing build step for 'libvterm'
TBL src/encoding/DECdrawing.tbl
TBL src/encoding/uk.tbl
CC src/encoding.c
make[3]: glibtool: No such file or directory
make[3]: *** [src/encoding.lo] Error 1
make[2]: *** [libvterm-prefix/src/libvterm-stamp/libvterm-build] Error 2
make[1]: *** [CMakeFiles/libvterm.dir/all] Error 2
make: *** [all] Error 2
/Users/jamie/.emacs.d/elpa/vterm-20201004.2057
-- System libvterm not found: libvterm will be downloaded and compiled as part of the build process
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/jamie/.emacs.d/elpa/vterm-20201004.2057/build
[  8%] Performing update step for 'libvterm'
[ 16%] No patch step for 'libvterm'
[ 25%] No configure step for 'libvterm'
[ 33%] Performing build step for 'libvterm'
CC src/encoding.c
CC src/keyboard.c
CC src/mouse.c
CC src/parser.c
CC src/pen.c
CC src/screen.c
CC src/state.c
CC src/unicode.c
src/unicode.c:215:12: warning: unused function 'mk_wcswidth' [-Wunused-function]
static int mk_wcswidth(const uint32_t *pwcs, size_t n)
           ^
src/unicode.c:306:12: warning: unused function 'mk_wcswidth_cjk' [-Wunused-function]
static int mk_wcswidth_cjk(const uint32_t *pwcs, size_t n)
           ^
2 warnings generated.
CC src/vterm.c
LINK libvterm.la
CC bin/unterm.c
CC bin/vterm-ctrl.c
bin/vterm-ctrl.c:117:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(csi);
         ^
bin/vterm-ctrl.c:117:10: note: did you mean 'strcmp'?
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:77:6: note: 'strcmp' declared here
int      strcmp(const char *__s1, const char *__s2);
         ^
bin/vterm-ctrl.c:117:10: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'char *' [-Wint-conversion]
  return strdup(csi);
         ^~~~~~~~~~~
bin/vterm-ctrl.c:142:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(dcs);
         ^
bin/vterm-ctrl.c:142:10: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'char *' [-Wint-conversion]
  return strdup(dcs);
         ^~~~~~~~~~~
bin/vterm-ctrl.c:129:10: warning: result of comparison of constant 156 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
    if(c == 0x9c) // ST
       ~ ^  ~~~~
3 warnings and 2 errors generated.
make[3]: *** [bin/vterm-ctrl] Error 1
make[2]: *** [libvterm-prefix/src/libvterm-stamp/libvterm-build] Error 2
make[1]: *** [CMakeFiles/libvterm.dir/all] Error 2
make: *** [all] Error 2
/Users/jamie/.emacs.d/elpa/vterm-20201004.2057
-- System libvterm not found: libvterm will be downloaded and compiled as part of the build process
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/jamie/.emacs.d/elpa/vterm-20201004.2057/build
[  8%] Performing update step for 'libvterm'
[ 16%] No patch step for 'libvterm'
[ 25%] No configure step for 'libvterm'
[ 33%] Performing build step for 'libvterm'
CC bin/vterm-ctrl.c
bin/vterm-ctrl.c:117:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(csi);
         ^
bin/vterm-ctrl.c:117:10: note: did you mean 'strcmp'?
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:77:6: note: 'strcmp' declared here
int      strcmp(const char *__s1, const char *__s2);
         ^
bin/vterm-ctrl.c:117:10: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'char *' [-Wint-conversion]
  return strdup(csi);
         ^~~~~~~~~~~
bin/vterm-ctrl.c:142:10: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  return strdup(dcs);
         ^
bin/vterm-ctrl.c:142:10: warning: incompatible integer to pointer conversion returning 'int' from a function with result type 'char *' [-Wint-conversion]
  return strdup(dcs);
         ^~~~~~~~~~~
bin/vterm-ctrl.c:129:10: warning: result of comparison of constant 156 with expression of type 'char' is always false [-Wtautological-constant-out-of-range-compare]
    if(c == 0x9c) // ST
       ~ ^  ~~~~
3 warnings and 2 errors generated.
make[3]: *** [bin/vterm-ctrl] Error 1
make[2]: *** [libvterm-prefix/src/libvterm-stamp/libvterm-build] Error 2
make[1]: *** [CMakeFiles/libvterm.dir/all] Error 2
make: *** [all] Error 2
/Users/jamie/.emacs.d/elpa/vterm-20201004.2057
```

**Comment by jeongsoolee09** [comment]
I ran into an exact same problem a few days ago, and managed to solve it myself. Please consult the thread I created (and closed): #403 .

**Comment by mpereira** [comment]
I'm seeing this as well after upgrading the vterm package.

**Comment by mpereira** [comment]
FTR, the `with-editor` package recently included [vterm support](https://github.com/magit/with-editor/commit/a339d6590096829a274c1901e407873e1fc1016e), which causes it to require `vterm` and trigger the problem reported in this issue.

**Comment by mpereira** [comment]
Jonas fixed the issue in `with-editor`: https://github.com/magit/with-editor/commit/6735180e73e787b79535c245b162249b70dbf841

**Comment by jixiuf** [comment]
this should be fixed after #404 is merged,
maybe you need set `(setq vterm-module-cmake-args "-DUSE_SYSTEM_LIBVTERM=no")`

**Comment by mpereira** [comment]
@jixiuf it was indeed fixed, thank you!

FTR I did not need to set `vterm-module-cmake-args`.
